libraries/AP_Math/quaternion.cpp

```diff
@@ -772,6 +772,20 @@ QuaternionT<T> &QuaternionT<T>::operator*=(const QuaternionT<T> &v)
     return *this;
 }
 
+// Note: Quaternions do not have a single, universally agreed-upon "division" definition like real numbers.
+// By definition, for a unit quaternion, its inverse equals its conjugate:
+//    q^-1 = conj(q) / |q|^2   (if |q|=1, then q^-1 = conj(q))
+// Quaternion multiplication is non-commutative, so the result of q1 / q2 can follow two conventions:
+// (1) Right division (commonly used in mathematics): q1 / q2 = q1 * q2^-1
+// (2) Left division (engineering convention): q1 / q2 = q2^-1 * q1
```

Comment:
I am unable to find citations on the names and common uses. E.g. the python libraries I can find use q1 * q2^-1.

I think the comment overall should briefly cover the following points:

* Division of q1/q2 can be q1 * q2^-1 or q2^-1 * q1
* The result in each case is different as quaternion multiplication is non-commutative
* We use the second definition after MATLAB